daemon/list.go

```diff
@@ -61,6 +64,21 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			}
 		}
 	}
+
+	if ancestors, ok := psFilters["ancestor"]; ok {
+		ancestorFilter = true
+		byParents := daemon.Graph().ByParent()
+		// The idea is to walk the graph down the most "efficient" way.
+		for _, ancestor := range ancestors {
+			// First, get the imageId of the ancestor filter (yay)
+			image, err := daemon.Repositories().LookupImage(ancestor)
+			if err == nil {
+				// Then walk down the graph and put the imageIds in imagesFilter
+				imagesFilter = append(imagesFilter, populateImageFilterByParents(image.ID, byParents)...)
```

Comment:
@duglin right :smile_cat:. And if it was a map it would be really simple to check. Let's try that :wink:.